### PR TITLE
Feature/uml

### PR DESCRIPTION
**Pull request by tsohleDev**
# Impliment UML to create class association

- Create a one-many relationship from ``Classroom`` to ``Student``
- Create a many-many relationship between ``Person`` and ``Book`` using ``Rental``

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (AnselemOdims) [inline book.rb]: - [x] Please kindly ensure the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` is created using the intermediate class `Rental` by creating an `add_rental` method.
- [x] Also, please ensure the `rental` has a getter and setter.
Hint:
```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
+  attr_accessor :name, :age
+
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Math.random(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+    super()
   end
```
Comment (AnselemOdims) [inline person.rb]: - [ ] Please kindly also implement the feedback for the `Book` class in the `Person` class.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @tsohleDev, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: book.rb (hunk 1 above)
Comment (tsohleDev) [reply book.rb]: Thank you

File: person.rb (hunk 2 above)
Comment (tsohleDev) [reply person.rb]: okay, thanks

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (cynthiainga) [inline book.rb]: - [x] As mentioned by the previous reviewer you should ensure that `rentals`  has a getter and setter which means the `rentals` should be `attr_accessor`, please kindly fix that 🙃 😁 
- [ ] Also is the case for the `person.rb` file, the `rentals` should have a getter and setter 😉 😉

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @tsohleDev, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.



 ## Highlights

- [ ] No linters errors ✔️ 
- [ ] Good commit history ✔️ 
- [ ] Descriptive PR ✔️ 

 ## Required Changes ♻️

- [ ] Please kindly check the comment under the review 👇 :poi]

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 3 above)
Comment (tsohleDev) [reply book.rb]: Thanks @cynthiainga

**Review by uwadonat — APPROVED** [review]
## Hi @tsohleDev,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Well organized and clean project ✔️
- Good communication with reviewers ✔️
thanks-awesome

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**